src/framework/preferences/js/LocalizationEnactor.js

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2018 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /*******************************************************************************
+     * Localization
+     *
+     * The enactor to change the locale shown according to the value
+     *
+     * This grade is resolvable from the root to allow for setting up of model relays
+     * from other components on a page that may want to be notified of a language
+     * change and update their own UI automatically.
+     *******************************************************************************/
+
+    fluid.defaults("fluid.prefs.enactor.localization", {
+        gradeNames: ["fluid.prefs.enactor", "fluid.contextAware", "fluid.resolveRoot"],
+        preferenceMap: {
+            "fluid.prefs.localization": {
+                "model.lang": "value"
+            }
+        },
+        contextAwareness: {
+            localeChange: {
+                checks: {
+                    inPanel: {
+                        contextValue: "{iframeRenderer}"
+                    },
+                    urlPath: {
```

Comment:
It seems the only localization method that the prefs framework supports is via the url path. If so, is it necessary to detect this case with the context awareness? Also, the default grade `fluid.prefs.enactor.localization.no-op` is not implemented. I'm curious how the localization would behave with this non-implemented grade.

Reply:
There are effectively two implemented methods, although an integrator could support additional ones. For example using a query parameter. The two methods supported are URL path and model. The no-op is essentially the model method as there is no additional functionality needed on our part. The integrator would have to hook into the model from the enactor, which is why it uses the "fluid.resolveRoot" grade to make it more easily discoverable outside of the context of the prefs framework's component tree structure.

Reply:
I think i just got what yo meant about the no-op grade. I believe that was something i was supposed to have removed from previous work. I've taken it out now.